instrumentation/nginx/src/otel_ngx_module.cpp

```diff
@@ -545,7 +545,14 @@ ngx_int_t FinishNgxSpan(ngx_http_request_t* req) {
   }
 
   auto span = context->request_span;
-  span->SetAttribute("http.status_code", req->headers_out.status);
+  const auto status_code = req->headers_out.status;
+  span->SetAttribute("http.status_code", status_code);
+
+  if (status_code >= 500) {
```

Comment:
Isn’t 400 also an error?

Reply:
4xx HTTP status code are client error code. I don't think it is fair to consider the whole span problematic when the issue is from the user.

Reply:
https://opentelemetry.io/docs/specs/semconv/http/http-spans/#status says you are right